rp2040-hal-examples/src/bin/spi_eh_bus.rs

```diff
@@ -0,0 +1,197 @@
+//! # SPI Bus example
+//!
+//! This application demonstrates how to construct a simple Spi Driver,
+//! and configure rp2040-hal's Spi peripheral to access it by utilising
+//! `ExclusiveDevice`` from `embedded_hal_bus`
+//!
+//!
+//! It may need to be adapted to your particular board layout and/or pin
+//! assignment.
+//!
+//! See the top-level `README.md` file for Copyright and license details.
+
+#![no_std]
+#![no_main]
+
+use embedded_hal::spi::Operation;
+use embedded_hal::spi::SpiDevice;
+use embedded_hal_bus::spi::ExclusiveDevice;
+// Ensure we halt the program on panic (if we don't mention this crate it won't
+// be linked)
+use panic_halt as _;
+
+use rp2040_hal::gpio::PinState;
+use rp2040_hal::uart::{DataBits, StopBits, UartConfig};
+// Alias for our HAL crate
+use rp2040_hal as hal;
+
+// Some traits we need
+use core::fmt::Write;
+use hal::clocks::Clock;
+use hal::fugit::RateExtU32;
+
+// A shorter alias for the Peripheral Access Crate, which provides low-level
+// register access
+use hal::pac;
+
+/// The linker will place this boot block at the start of our program image. We
+/// need this to help the ROM bootloader get our code up and running.
+/// Note: This boot block is not necessary when using a rp-hal based BSP
+/// as the BSPs already perform this step.
+#[link_section = ".boot2"]
+#[used]
+pub static BOOT2: [u8; 256] = rp2040_boot2::BOOT_LOADER_GENERIC_03H;
+
+/// External high-speed crystal on the Raspberry Pi Pico board is 12 MHz. Adjust
+/// if your board has a different frequency
+const XTAL_FREQ_HZ: u32 = 12_000_000u32;
+
+/// Our Spi device driver
+/// We need to use a generic here (SPI), because we want our driver to work for any other
+/// microcontroller that implements the embedded-hal Spi traits
+struct MySpiDriver<SPI> {
+    spi: SPI,
+}
+
+/// When things go wrong, we could return Error(()) but that wouldn't help anyone troubleshoot
+/// so we'll create an error type with additional info to return.
+#[derive(Copy, Clone, Debug)]
+enum MyError<SPI> {
```

Comment:
Maybe call this type parameters or ERR?

Reply:
I'm not sure I understand which "this" you're referring to with options like that.

Reply:
Sorry. That my fault for using my phone.

Can you change the SPI type parameter to E or ERR?

Reply:
I see where you were going with this, E is an improvement